### PR TITLE
Fix grid solver not updating variables when height/width changed

### LINKED ISSUE #2093: PanzoomCamera produces an un-invertable matrix
Tests on CI are now failing with:

```
   File "/home/runner/work/vispy/vispy/vispy/scene/cameras/panzoom.py", line 195, in viewbox_resize_event
    self.view_changed()
  File "/home/runner/work/vispy/vispy/vispy/scene/cameras/base_camera.py", line 437, in view_changed
    self._update_transform()
  File "/home/runner/work/vispy/vispy/vispy/scene/cameras/panzoom.py", line 291, in _update_transform
    self.tf_mat.matrix = self.transform.as_matrix().matrix
  File "/home/runner/work/vispy/vispy/vispy/visuals/transforms/linear.py", line 219, in as_matrix
    m.scale(self.scale)
  File "/home/runner/work/vispy/vispy/vispy/visuals/transforms/linear.py", line 445, in scale
    self.matrix = np.dot(self.matrix, scale)
  File "/home/runner/work/vispy/vispy/vispy/visuals/transforms/linear.py", line 401, in matrix
    self.shader_imap()
  File "/home/runner/work/vispy/vispy/vispy/visuals/transforms/linear.py", line 389, in shader_imap
    fn['inv_matrix'] = self.inv_matrix  # uniform mat4
  File "/home/runner/work/vispy/vispy/vispy/visuals/transforms/linear.py", line 407, in inv_matrix
    self._inv_matrix = np.linalg.inv(self.matrix)
  File "<__array_function__ internals>", line 6, in inv
  File "/home/runner/micromamba/envs/vispy-tests/lib/python3.7/site-packages/numpy/linalg/linalg.py", line 545, in inv
    ainv = _umath_linalg.inv(a, signature=signature, extobj=extobj)
  File "/home/runner/micromamba/envs/vispy-tests/lib/python3.7/site-packages/numpy/linalg/linalg.py", line 88, in _raise_linalgerror_singular
    raise LinAlgError("Singular matrix")
numpy.linalg.LinAlgError: Singular matrix
```

### PR DESCRIPTION
**Pull request by djhoese**
Closes #2093 
Closes #2099 
Closes #2097 

This closes the issue mentioned above and replaces the PRs mentioned. While debugging #2093 I noticed that the `height` of a widget would sometimes be set to `0` randomly. I thought I remembered kiwisolver's documentation talking about how to print out information about the solver so I was looking for [that](https://kiwisolver.readthedocs.io/en/latest/basis/solver_internals.html#inspecting-the-solver-state) when I found https://kiwisolver.readthedocs.io/en/latest/basis/basic_systems.html#solving-and-updating-variables. This mentions that if you suggest a new value for a variable you also have to later run `.updateVariables()` to have the solver actually calculate the new results.

I made the changes in this PR and ran the example code in #2093 for 438 times with no errors. This seems to actually fix the issue. I'm hoping @almarklein can do a little more testing. It would also be interesting (if we have the time) to know what the previous height was and what the new height was being set to. I guess I assume the height is being set to 0 originally, then the whole window is resized during `.show`, and the height never got changed from 0...well except why would width not be the problem :thinking: I don't know.

### CODE REVIEW

**Comment by almarklein** [comment]
Another 700 runs without errors.

File: vispy/scene/widgets/grid.py [hunk 1]
```diff
@@ -496,6 +492,7 @@ def _update_child_widget_dim(self):
                 y = np.sum(value_vectorized(self._height_grid[col][0:row]))
 
             if isinstance(widget, ViewBox):
+                print("####", val, x, y, width, height)
```
Comment (almarklein) [inline vispy/scene/widgets/grid.py:495]: Lets remove this print statement.

**Review by almarklein — APPROVED** [review]

File: vispy/scene/widgets/grid.py (hunk 1 above)
Comment (djhoese) [reply vispy/scene/widgets/grid.py]: That's what I get for making a PR right before bed. Done.

**Comment by almarklein** [comment]
This looks fine to me, but I don't really know how the solver works so I can't review any deeper than confirming that I don't see anything odd (or leftover print-statements). Feel free to merge.

**Comment by almarklein** [comment]
Try a squash-merge :)

**Pull request merged**